src/gui/annotations/qgsmaptoolselectannotation.cpp

```diff
@@ -0,0 +1,694 @@
+    }
+    else if ( event->key() == Qt::Key_Left )
+    {
+      deltaX = -pixels;
+    }
+    else if ( event->key() == Qt::Key_Right )
+    {
+      deltaX = pixels;
+    }
+
+    for ( auto &selectedItem : mSelectedItems )
+    {
+      selectedItem->attemptMoveBy( deltaX, deltaY );
+    }
+    event->ignore();
+  }
+}
+
+QList<QgsAnnotationItemRubberBand *> QgsMapToolSelectAnnotation::selectedItems() const
+{
+  QList<QgsAnnotationItemRubberBand *> items;
+  for ( auto &selectedItem : mSelectedItems )
+  {
+    items << selectedItem.get();
+  }
+  return items;
+}
+
+void QgsMapToolSelectAnnotation::onCanvasRefreshed()
+{
+  const QgsRenderedItemResults *renderedItemResults = canvas()->renderedItemResults( false );
+  if ( !renderedItemResults )
+  {
+    return;
+  }
+
+  const QList<QgsRenderedItemDetails *> items = renderedItemResults->renderedItems();
+  bool needsSelectedItemsUpdate = false;
+  for ( auto &selectedItem : mSelectedItems )
+  {
+    if ( selectedItem->needsUpdatedBoundingBox() )
+    {
+      needsSelectedItemsUpdate = true;
+
+      auto it = std::find_if( items.begin(), items.end(), [&selectedItem]( const QgsRenderedItemDetails *item ) {
+        if ( const QgsRenderedAnnotationItemDetails *annotationItem = dynamic_cast<const QgsRenderedAnnotationItemDetails *>( item ) )
+        {
+          if ( annotationItem->itemId() == selectedItem->itemId() && annotationItem->layerId() == selectedItem->layerId() )
+          {
+            return true;
+          }
+        }
+        return false;
+      } );
+
+      if ( it != items.end() )
+      {
+        selectedItem->updateBoundingBox( ( *it )->boundingBox() );
+      }
+    }
+  }
+
+  if ( needsSelectedItemsUpdate )
+  {
+    emit selectedItemsChanged();
+  }
+}
+
+const QgsRenderedAnnotationItemDetails *QgsMapToolSelectAnnotation::findClosestItemToPoint( const QgsPointXY &mapPoint, const QList<const QgsRenderedAnnotationItemDetails *> &items, QgsRectangle &bounds )
+{
+  const QgsRenderedAnnotationItemDetails *closestItem = nullptr;
+  double closestItemDistance = std::numeric_limits<double>::max();
+  double closestItemArea = std::numeric_limits<double>::max();
+
+  for ( const QgsRenderedAnnotationItemDetails *item : items )
+  {
+    const QgsAnnotationItem *annotationItem = annotationItemFromId( item->layerId(), item->itemId() );
+    if ( !annotationItem )
+      continue;
+
+    const QgsRectangle itemBounds = item->boundingBox();
+    const double itemDistance = itemBounds.contains( mapPoint ) ? 0 : itemBounds.distance( mapPoint );
+    if ( !closestItem || itemDistance < closestItemDistance || ( itemDistance == closestItemDistance && itemBounds.area() < closestItemArea ) )
+    {
+      closestItem = item;
+      closestItemDistance = itemDistance;
+      closestItemArea = itemBounds.area();
+      bounds = itemBounds;
+    }
+  }
+  return closestItem;
+}
+
+QgsAnnotationLayer *QgsMapToolSelectAnnotation::annotationLayerFromId( const QString &layerId )
```

Comment:
Same with all this -- should be a moved to a common "QgsAnnotationMapTool" base class